data-transfer/src/main/java/it/eng/datatransfer/service/api/RestArtifactService.java

```diff
@@ -24,63 +20,52 @@
-	private final CredentialUtils credentialUtils;
-	private final DataTransferProperties properties;
+	private final ApplicationEventPublisher publisher;
 	
-	public RestArtifactService(ApplicationEventPublisher publisher, DataTransferService dataTransferService, MongoTemplate mongoTemplate,
-			OkHttpRestClient okHttpRestClient, CredentialUtils credentialUtils, DataTransferProperties properties) {
+	public RestArtifactService(DataTransferService dataTransferService, MongoTemplate mongoTemplate,
+			OkHttpRestClient okHttpRestClient, ApplicationEventPublisher publisher) {
 		super();
-		this.publisher = publisher;
 		this.dataTransferService = dataTransferService;
 		this.mongoTemplate = mongoTemplate;
 		this.okHttpRestClient = okHttpRestClient;
-		this.credentialUtils = credentialUtils;
-		this.properties = properties;
+		this.publisher = publisher;
 	}
 
-	public String getArtifact(String transactionId, JsonNode jsonBody) {
-		TransferProcess transferProcess = getTransferProcessForTransactionId(transactionId);
-		log.info("Publishing event to increase counter for agreementId {}", transferProcess.getAgreementId());
-		publisher.publishEvent(new ArtifactConsumedEvent(transferProcess.getAgreementId()));
-		return getJohnDoe();
-	}
-	
 	public GridFsResource streamAttachment(String transactionId) {
 		TransferProcess transferProcess = getTransferProcessForTransactionId(transactionId);
-		GenericApiResponse<String> response = okHttpRestClient.sendGETRequest("http://localhost:" + properties.serverPort() + ApiEndpoints.CATALOG_DATASETS_V1 + "/fileid", 
-				credentialUtils.getAPICredentials());
+		String response = okHttpRestClient.sendInternalRequest(ApiEndpoints.CATALOG_DATASETS_V1 + "/" + transferProcess.getDatasetId() + "/fileid", HttpMethod.GET, null);
+		
+		GenericApiResponse<String> rr = Serializer.deserializePlain(response, GenericApiResponse.class);
 		
-		String fileId = response.getData();
+		String fileId = rr.getData();
 		if(StringUtils.isBlank(fileId)) {
 			log.error("NO file attached to dataset");
 			throw new TransferProcessArtifactNotFoundException("Artifact not found for agreement " + transferProcess.getAgreementId(), 
 					transferProcess.getConsumerPid(), transferProcess.getProviderPid());
 		}
 	 	GridFSBucket gridFSBucket = GridFSBuckets.create(mongoTemplate.getDb());
-	 	Bson query = Filters.eq("_id", fileId);
+	 	ObjectId fileIdentifier = new ObjectId(fileId);
+	 	Bson query = Filters.eq("_id", fileIdentifier);
 	 	GridFSFile file = gridFSBucket.find(query).first();
         if (file != null) {
             GridFSDownloadStream gridFSDownloadStream = gridFSBucket.openDownloadStream(file.getObjectId());
             GridFsResource gridFsResource = new GridFsResource(file, gridFSDownloadStream);
-            publisher.publishEvent(new ArtifactConsumedEvent(transferProcess.getAgreementId()));
+    		publisher.publishEvent(new ArtifactConsumedEvent(transferProcess.getAgreementId()));
```

Comment:
This one here is not most elegant place, we can for example, increase counter by 1 and in controller it can break something.
Not sure if we can create some method in service to publish event and call it from controller in finally block IF NO EXCEPTION occurs.